File: amigotest/src/main/java/amigo/tn/entities/Reclamation.java
package amigo.tn.entities;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import amigo.tn.security.user.User;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Data
@Getter
@Setter
@Document(collection = "reclamations")
public class Reclamation {
    @Id
    private String id;
    private String titre;
    private String description;
    private String auteur;
    private Date dateCreation;
    private Date dateModification;
    private StatutReclamation statut;
    @DBRef
    private User user;
}

File: amig/src/app/entities/reclamation.ts
import { User } from "./user";

export class Reclamation {
    id: string;
    titre: string;
    description: string;
    dateCreation: Date;
    dateModification: Date;
    statut: StatutReclamation;
    user: User;
  }
  
  export enum StatutReclamation {
    EN_ATTENTE = 'EN_ATTENTE',
    EN_COURS = 'EN_COURS',
    TERMINEE = 'TERMINEE',
    ANNULEE = 'ANNULEE'
  }